opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/CalciteScriptEngine.java

```diff
@@ -0,0 +1,300 @@
+      String referenceField = expression.getReferenceForTermQuery();
+      if (StringUtils.isEmpty(referenceField)) {
+        throw new UnsupportedScriptException(
+            "Field name cannot be empty for expression: " + expression);
+      }
+      MethodCallExpression fieldValueExpr =
+          Expressions.call(
+              DataContext.ROOT,
+              BuiltInMethod.DATA_CONTEXT_GET.method,
+              Expressions.constant(expression.getReferenceForTermQuery()));
+      if (storageType == null) {
+        final RelDataType fieldType = rowType.getFieldList().get(index).getType();
+        storageType = ((JavaTypeFactory) typeFactory).getJavaClass(fieldType);
+      }
+      return EnumUtils.convert(tryConvertDocValue(fieldValueExpr, exprType), storageType);
+    }
+
+    /**
+     * DocValue only support long and double for integer and float, cast to the related type first
+     */
+    private Expression tryConvertDocValue(Expression docValueExpr, ExprType exprType) {
+      return switch (exprType) {
+        case INTEGER -> EnumUtils.convert(docValueExpr, Long.class);
+        case FLOAT -> EnumUtils.convert(docValueExpr, Double.class);
+        default -> docValueExpr;
+      };
+    }
+  }
+
+  public static class ScriptDataContext implements DataContext {
+
+    private final Supplier<Map<String, ScriptDocValues<?>>> docProvider;
+    private final Map<String, Object> params;
+
+    public ScriptDataContext(
+        Supplier<Map<String, ScriptDocValues<?>>> docProvider, Map<String, Object> params) {
+      this.docProvider = docProvider;
+      this.params = params;
+    }
+
+    @Override
+    public @Nullable SchemaPlus getRootSchema() {
+      return null;
+    }
+
+    @Override
+    public JavaTypeFactory getTypeFactory() {
+      return null;
+    }
+
+    @Override
+    public QueryProvider getQueryProvider() {
+      return null;
+    }
+
+    @Override
+    public Object get(String name) {
+      // UTC_TIMESTAMP is a special variable used for some time related functions.
+      if (Variable.UTC_TIMESTAMP.camelName.equals(name))
+        return params.get(Variable.UTC_TIMESTAMP.camelName);
+
+      ScriptDocValues<?> docValue = docProvider.get().get(name);
+      if (docValue == null || docValue.isEmpty()) {
+        return null; // No way to differentiate null and missing from doc value
+      }
+
+      Object value = docValue.get(0);
+      if (value instanceof ChronoZonedDateTime) {
+        // We store timestamp as string in the current implementation with Calcite.
+        // And the string should have the format defined in ExprTimestampValue
+        // TODO: should we change to store timestamp as Instant in the future.
+        return new ExprTimestampValue(((ChronoZonedDateTime<?>) value).toInstant()).value();
+      }
+      return value;
+    }
+  }
+
+  /**
+   * This function is copied from Calcite RexExecutorImpl It's used to compile RexNode expression to
```

Comment:
Seems you copied the decompiled code instead source code. Can you check it?:
`java.util.Iterator var5`, `Expressions.declare(16`, `Expressions.methodDecl(1`, etc.